src/cursor/cur_table.c

```diff
@@ -918,9 +919,24 @@ __curtable_open_indices(WT_CURSOR_TABLE *ctable)
 
     WT_RET(__wt_calloc_def(session, table->nindices, &ctable->idx_cursors));
     for (i = 0, cp = ctable->idx_cursors; i < table->nindices; i++, cp++)
-        WT_RET(
+        WT_ERR(
           __wt_open_cursor(session, table->indices[i]->source, &ctable->iface, ctable->cfg, cp));
-    return (0);
+
+    if (0) {
+err:
+        /*
```

Comment:
This is all good. However you need to add a similar handling for colgroups in the function right above this one. And please add a colgroup test case to your new test.

Reply:
It turns out that colgroups are treated differently, they are opened when the table cursor is opened.  If column group open fails, the cursor open fails, so you can't get a "partially opened" cursor.